tests/test_diffraction_objects.py

```diff
@@ -487,13 +427,39 @@ def test_dump(tmp_path, mocker):
 ]
 
 
-@pytest.mark.parametrize("inputs, expected", tc_params)
-def test_constructor(inputs, expected):
-    actual = DiffractionObject(**inputs).__dict__
-    diff = DeepDiff(actual, expected, ignore_order=True, significant_digits=13, exclude_paths="root['_id']")
+@pytest.mark.parametrize(
+    "init_args, expected_do_dict",
+    test_init_valid_params,
+)
+def test_init_valid(init_args, expected_do_dict):
+    actual_do_dict = DiffractionObject(**init_args).__dict__
+    diff = DeepDiff(
+        actual_do_dict, expected_do_dict, ignore_order=True, significant_digits=13, exclude_paths="root['_id']"
+    )
     assert diff == {}
 
 
+test_init_invalid_params = [
```

Comment:
`xarray` `yarrays` and `xtype` requires - test cases when these are not provided during `init`